sync/git.go

```diff
@@ -0,0 +1,91 @@
+package sync
+
+import (
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing/object"
+	"github.com/knqyf263/pet/config"
+	"github.com/pkg/errors"
+	"io/ioutil"
+	"path/filepath"
+	"time"
+)
+
+type GitClient struct{}
+
+func NewGitClient() (Client, error) {
+	return GitClient{}, nil
+}
+
+func (g GitClient) GetSnippet() (*Snippet, error) {
+	directory, err := getDirectory()
+	r, err := getRepository(directory)
+	w, err := r.Worktree()
+
+	err = w.Pull(&git.PullOptions{RemoteName: "origin"})
+	filename := filepath.Join(directory, "snippet.toml")
+	ref, err := r.Head()
+	commit, err := r.CommitObject(ref.Hash())
+	content, err := ioutil.ReadFile(filename)
+
+	if err != nil {
+		return nil, errors.Wrapf(err, "snippet.toml")
+	}
+
+	return &Snippet{
+		Content:   string(content),
+		UpdatedAt: commit.Author.When,
+	}, nil
+}
+
+func (g GitClient) UploadSnippet(body string) error {
+	directory, err := getDirectory()
+	author, authorEmail, err := getGitSignature()
+	r, err := getRepository(directory)
+	w, err := r.Worktree()
+
+	if err != nil {
+		return errors.Wrapf(err, "Failed to load repository")
+	}
+
+	_, err = w.Add("snippet.toml")
+	_, err = w.Commit("update snippets", &git.CommitOptions{
+		Author: &object.Signature{
+			Name:  author,
+			Email: authorEmail,
+			When:  time.Now(),
+		},
+	})
+
+	return r.Push(&git.PushOptions{})
+}
+
+func getDirectory() (string, error) {
+	if config.Conf.Git.Directory != "" {
+		return config.Conf.Git.Directory, nil
+	}
+	return "", errors.New("Git Directory not found")
```

Comment:
https://github.com/golang/go/wiki/CodeReviewComments#error-strings

```suggestion
	return "", errors.New("git directory not found")
```